android/app/build.gradle

```diff
@@ -47,7 +47,7 @@ android {
         applicationId "com.example.flutter_minimal_app"
         // You can update the following values to match your application needs.
         // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-build-configuration.
-        minSdkVersion flutter.minSdkVersion
+        minSdkVersion 21
         targetSdkVersion flutter.targetSdkVersion
```

Comment:
Why only update minSdkversion? By default after flutter 2.8, these values are created in android/local.properties file, but since that file is git ignored, these values will be empty when building in CI. 
Also, looks like compileSdkVerison is missing? not sure how it goes with Flutter, but with react-native min app I had seen the issue "compileSdkVersion is not specified, please add to build.gradle".

Reply:
I have updated targetSdkVersion and compileSdk version to 31